### DISCUSSION TITLE
Select previous command output without shell integration

### DISCUSSION THREAD

**Discussion by tambourine-man** [wezterm/wezterm#6113] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I'm loving WezTerm and considering switching to it. 

On the Mac's Terminal.app, if “Automatically Mark Prompt Lines” is selected, I can hit `CMD-SHIFT-A` (“Select Between Marks”) and copy the previous command output. I understand that I can do the same with WezTerm by using shell integration, but I don't want to install it in every server I SSH into.

Most of the time the output of `pwd` or `realpath` is what I'm interested in, so simply selecting the previous line would work fine as well.

Quick Select almost fits the bill, except selecting the last line is not always done by hitting the same letter and I'm a creature of habit :)

In summary, if I could assign a key combo that would select the previous line and copy it I would be set.
Thanks a lot

**Comment by bew** [comment]
You might achieve what you want with some Lua config using:
- https://wezfurlong.org/wezterm/config/lua/wezterm/action_callback.html
- https://wezfurlong.org/wezterm/config/lua/pane/get_logical_lines_as_text.html
- and then copy the line you want 🤔
